rlp/list.libevm.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package rlp
+
+// InList is a convenience wrapper, calling `fn` between calls to
+// [EncoderBuffer.List] and [EncoderBuffer.ListEnd]. If `fn` returns an error,
+// it is propagated directly.
+func (b EncoderBuffer) InList(fn func() error) error {
+	l := b.List()
+	if err := fn(); err != nil {
+		return err
+	}
+	b.ListEnd(l)
```

Comment:
supernit perhaps rename `l` to `i`, since it's a list index?

```suggestion
	i := b.List()
	if err := fn(); err != nil {
		return err
	}
	b.ListEnd(i)
```